File: BoardGames/HitOrMiss/include/libpython/score_controller.py
import score_model
import score_view

score = None

def initialize_score() :
  global score

  score = score_model.create_score()

def show_score() :
  if score :
    lines = score_view.build_score_lines( score )

    for line in lines :
      print line

    return lines
  else :
    return None

def store_games() :
  global score

  score = score_model.add_games( score )

def store_hits() :
  global score

  score = score_model.add_hits( score )

def store_misses() :
  global score

  score = score_model.add_misses( score )

File: BoardGames/HitOrMiss/include/libruby/score_controller.rb
require_relative 'score_model'
require_relative 'score_view'

$score = nil

def initialize_score()
  $score = create_score
end

def show_score()
  score_lines = build_score_lines( $score )

  if not score_lines.nil?
    score_lines.each do | line |
      puts line
    end
  end

  return score_lines
end

def store_games()
  $score = add_games( $score )
end

def store_hits()
  $score = add_hits( $score )
end

def store_misses()
  $score = add_misses( $score )
end
